pkg/controller/appdefinition/volume.go

```diff
@@ -155,7 +156,31 @@ func toPVCs(req router.Request, appInstance *v1.AppInstance) (result []kclient.O
 
 		if bind {
 			pvc.Name = bindName(vol)
-			pvc.Spec.VolumeName = volumeBinding.Volume
+
+			// We need to get the real PersistentVolume that will be bound.
+			// Its public name is volumeBinding.Volume, so we will use that in the label selector.
+			// In subsequent calls to this same function, the public name label will have been changed to the new AppInstance,
+			// so we also need to account for that in the label selector.
+			requirement, err := klabels.NewRequirement(labels.AcornPublicName, selection.In, []string{
+				volumeBinding.Volume,
+				publicname.ForChild(appInstance, volumeBinding.Target),
+			})
+			if err != nil {
+				return nil, err
+			}
+			selector := klabels.SelectorFromSet(map[string]string{
+				labels.AcornAppNamespace: appInstance.Namespace,
+			}).Add(*requirement)
+
+			pvList := new(corev1.PersistentVolumeList)
+			if err := req.Client.List(req.Ctx, pvList, &kclient.ListOptions{LabelSelector: selector}); err != nil {
+				return nil, err
+			}
+			if len(pvList.Items) != 1 {
+				return nil, fmt.Errorf("expected 1 PV for volume %s, found %d", volumeBinding.Volume, len(pvList.Items))
+			}
+
+			pvc.Spec.VolumeName = pvList.Items[0].Name
```

Comment:
Spoke "offline" about this. These changes make it impossible for a user to bind an existing volume created outside of Acorn. We discussed the approach of how to handle these two cases.

Reply:
Didn't realize we supported that! Are we sure we want to? You have any concrete use cases in mind?

Reply:
It is essentially the same thing that happens [here](https://www.acorn.io/hosting-the-pi-hole-ad-blocker-on-my-kubernetes-home-lab/) except that you could bind to a volume with a different name.